### ISSUE TITLE
Begin automatic documentation

### ISSUE THREAD

**Issue by paddy-r** [paddy-r/StrEmbed-6-1#11]
Create doc folder once new file structure established (see issue #3). Ultimate goal is to create self-documenting code in parallel with manual and readme.

* [ ] Start with doc folder
* [ ] Add some docstrings to step_parse for starters
* [ ] Choose output format via Sphinx
* [ ] Check deployment (via Github Pages or Read the Docs -> decide!)
* [ ] Add anything to .gitignore to tidy up
* [ ] Draw line when output looks good; then raise another issue to streamline readme, manual and automated (code) documentation into fewer processes/outputs
